jetstream/tools/maxtext/model_ckpt_conversion.sh

```diff
@@ -0,0 +1,81 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -ex
+
+idx=$(date +%Y-%m-%d-%H-%M)
+# Modify the `MODEL` and `MODEL_VARIATION` based on the model you use.
+export MODEL=$1
+export MODEL_VARIATION=$2
+export MODEL_NAME=${MODEL}-${MODEL_VARIATION}
+
+# After downloading checkpoints, copy them to GCS bucket at $CHKPT_BUCKET \
+# Please use seperate GCS paths for uploading model weights from kaggle ($CHKPT_BUCKET) and MaxText compatible weights ($MODEL_BUCKET).
+# Point these variables to a GCS bucket that you created.
+export CHKPT_BUCKET=gs://${USER}-maxtext/chkpt/${MODEL}/${MODEL_VARIATION}
+export MODEL_BUCKET=gs://${USER}-maxtext
+
+# Point `BASE_OUTPUT_DIRECTORY` to a GCS bucket that you created, this bucket will store all the files generated by MaxText during a run.
+export BASE_OUTPUT_DIRECTORY=gs://${USER}-runner-maxtext-logs
+
+# Point `DATASET_PATH` to the GCS bucket where you have your training data
+export DATASET_PATH=gs://${USER}-maxtext-dataset
+
+export BUCKET_LOCATION=US
+
+# Create three GCS buckets for the demo.
+gcloud storage buckets create ${MODEL_BUCKET} --location=${BUCKET_LOCATION} || true
+gcloud storage buckets create ${BASE_OUTPUT_DIRECTORY} --location=${BUCKET_LOCATION} || true
+gcloud storage buckets create ${DATASET_PATH} --location=${BUCKET_LOCATION} || true
+
+# Copy the downloaded checkpoints to `CHKPT_BUCKET`.
+# Gemma example: gsutil -m cp -r 7b ${CHKPT_BUCKET}
+# Llama2 example: gsutil -m cp -r llama-2-7b ${CHKPT_BUCKET}
+gsutil -m cp -r $3 ${CHKPT_BUCKET}
+
+# Covert model checkpoints to MaxText compatible checkpoints.
+if [ "$MODEL" == "gemma" ]; then
+    CONVERT_CKPT_SCRIPT="convert_gemma_chkpt.py"
+else
+    # We install torch CPU because the checkpoint conversion script MaxText/llama_or_mistral_ckpt.py does not need a TPU/GPU
+    pip install torch --index-url https://download.pytorch.org/whl/cpu
+    CONVERT_CKPT_SCRIPT="llama_or_mistral_ckpt.py"
+fi
+
+JAX_PLATFORMS=cpu python MaxText/${CONVERT_CKPT_SCRIPT} \
+  --base-model-path ${CHKPT_BUCKET} \
+  --maxtext-model-path ${MODEL_BUCKET}/${MODEL}/${MODEL_VARIATION}/${idx} \
+  --model-size ${MODEL_VARIATION}
+echo "Writen MaxText compatible checkpoint to ${MODEL_BUCKET}/${MODEL}/${MODEL_VARIATION}/${idx}"
+
+# We define `CONVERTED_CHECKPOINT` to refer to the checkpoint subdirectory.
+export CONVERTED_CHECKPOINT=${MODEL_BUCKET}/${MODEL}/${MODEL_VARIATION}/${idx}/0/items
+
+# Covert MaxText compatible checkpoints to unscanned checkpoints.
```

Comment:
We should also add some details on what kinds of resources are needed to convert (or finetune) checkpoint. For llama2-13B/70b, it needed a v4-128 to get it done. This can go into README/User guide as well.

Reply:
For checkpoint conversion, I enabled `JAX_PLATFORMS=cpu` here, so we don't require TPU. But for AQT fine tuning, we have TPU requirements. Updated description in the scripts and user guide.